admin/includes/functions/html_output.php

```diff
@@ -206,6 +206,9 @@ function zen_draw_input_field($name, $value = '~*~*#', $parameters = '', $requir
 
     $field .= ' />';
 
+    if ($required) {
+      $field .= '&nbsp;<span class="alert">' . TEXT_FIELD_REQUIRED . '</span>';

```

Comment:
To maintain html validation, shouldn't `TEXT_FIELD_REQUIRED` be tested to have/contain something before adding the `<span>`?

Reply:
An empty span is not a validation error.

Reply:
No not an error, but an unnecessary warning.

Reply:
Why would you validate the admin side?  And why would you blank out this define?  This seems excessive.

Reply:
A non-validated admin side page may have problems executing javascript/jQuery.  Validation nowadays can be done within an individual's browser and not require reaching out to provide information/content from the page. 

May be blanked out on either the admin or catalog side (I chose not to duplicate the comments at PR #2589 after first having seen and commenting on this) to instead support some form of such javascript/jQuery or perhaps that all fields have been made required and the excess identifier is found unnecessary, etc...

A few recent examples of checking validation (notably on the admin) include PRs #2572, #2573, and #2513.